ionic-team/ionicons
# chore: SVG optimization no longer breaks outline icons

## Issue

Certain outline icons are being rendered as filled.

## Why it's happening

The SVG optimization script currently causes some outline SVGs to be filled. SVGs with `fill="none"` set on the `svg` element has this attribute replaced by an `ionicon-fill-none` class: https://github.com/ionic-team/ionicons/blob/116d2502556ed580bd4899d52cb956e8588c5691/scripts/build.ts#L58

However, the optimized SVG template is then updated to a) remove all non-essential attributes from the `svg` element and b) add the `ionicon` class: https://github.com/ionic-team/ionicons/blob/116d2502556ed580bd4899d52cb956e8588c5691/scripts/build.ts#L170-L171

This causes the `ionicon-fill-none` class to be removed from the `svg` element. As a result, outline icons appear filled.

## How I fixed it

This PR fixes the issue by removing the `.replace` call. During a previous optimization step I ensure that the `ionicon` class is added to the `svg` element. There are other attributes that need to be removed such as `width` and `height`. However, these are already being removed in https://github.com/ionic-team/ionicons/blob/116d2502556ed580bd4899d52cb956e8588c5691/scripts/build.ts#L79.

## How to test this PR

1. Clone the repo and check out `icon-binocs`. This is icon is being added in https://github.com/ionic-team/ionicons/pull/1258 and reproduces this bug.
2. Run `npm run build`.
3. Open `dist/svg/binoculars-outline.svg` and verify that the icon appears filled.
4. Merge that branch with this fix. `git fetch && git merge origin/icon-svg-optimization`.
5. Run `npm run build`.
6. Open `dist/svg/binoculars-outline.svg` and verify that the icon appears outlined.

| Before | After |
| - | - |
| <img src="https://github.com/ionic-team/ionicons/assets/2721089/1725bf45-4ced-4b0f-9c35-39e11b936493" width="200" /> | <img src="https://github.com/ionic-team/ionicons/assets/2721089/e6f512a7-a8d1-495b-9321-3a12f1c1e5d7" width="200" /> |
